### PR TITLE
[tests-only][full-ci]Add custom permission e2e tests

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description

This Pr refactor codes related to custom permission for resource to work for file.

## Related Issue

- https://github.com/owncloud/web/issues/8101

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/31260/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31260/tracing/file-alice-2023-1-9-07-39-10.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31260/tracing/file-brian-2023-1-9-07-39-20.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31260/tracing/upload-resources-in-personal-space-alice-2023-1-9-07-40-45.zip 


</p></details>

File: tests/e2e/cucumber/features/smoke/share.oc10.feature [hunk 1]
```diff
@@ -77,15 +85,18 @@ Feature: share
     And "Alice" uploads the following resource
       | resource        | to               |
       | testavatar.jpeg | folder_to_shared |
+      | lorem.txt           |                  |
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/share.oc10.feature:88]: what will happen if the `to` is empty?
```suggestion
      | testavatar.jpeg | folder_to_shared |
      | lorem.txt       |                  |
```

File: tests/e2e/cucumber/features/smoke/tags.ocis.feature [hunk 2]
```diff
@@ -66,8 +66,8 @@ Feature: Users can use web to organize tags
       | resource                   | tags         |
       | folder_to_shared/lorem.txt | tag 1, tag 2 |
     When "Alice" shares the following resource using the sidebar panel
-      | resource         | recipient | type | role   |
-      | folder_to_shared | Brian     | user | editor |
+      | resource         | recipient | type | role   | resourceType |
+      | folder_to_shared | Brian     | user | editor |folder         |
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/tags.ocis.feature:70]:
```suggestion
      | resource         | recipient | type | role   | resourceType |
      | folder_to_shared | Brian     | user | editor |folder        |
```

File: tests/e2e/support/objects/app-files/share/collaborator.ts [hunk 3]
```diff
@@ -93,31 +100,41 @@ export default class Collaborator {
...
     // the role of the first collaborator is used as the collaborators role
     const role = collaborators[0].role
+    const resourceType = collaborators[0].resourceType
 
     for (const collaborator of collaborators) {
       await Collaborator.addCollaborator({ page, collaborator })
     }
-    await Collaborator.setCollaboratorRole(page, role)
+    await Collaborator.setCollaboratorRole(page, role, resourceType)
+    // await Collaborator.setCollaboratorRole(page, role)
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/share/collaborator.ts:109]:
```suggestion
    await Collaborator.setCollaboratorRole(page, role, resourceType)
```

File: tests/e2e/support/objects/app-files/share/collaborator.ts [hunk 4]
```diff
@@ -93,31 +100,41 @@ export default class Collaborator {
...
   static async setCustomPermissions(
     page: Page,
-    permissions: CustomPermissionType[]
+    permissions: CustomPermissionType[],
+    resourceType: string
   ): Promise<void> {
+    let CUSTOM_PERMISSIONS: readonly CustomPermissionType[] = Collaborator.FILE_CUSTOM_PERMISSIONS
+    if (resourceType === 'folder') {
+      CUSTOM_PERMISSIONS = Collaborator.FOLDER_CUSTOM_PERMISSIONS
+    }
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/share/collaborator.ts:121]:
```suggestion
    const CUSTOM_PERMISSIONS: CustomPermissionType[] = resourceType === 'folder' ? CUSTOM_PERMISSIONS = Collaborator.FOLDER_CUSTOM_PERMISSIONS : Collaborator.FILE_CUSTOM_PERMISSIONS
```

**Comment by saw-jan** [comment]
@amrita-shrestha, target stable branch first.

**Comment by amrita-shrestha** [comment]
> @amrita-shrestha, target stable branch first.

when i started this PR all changes was not present in stable so i did my PR in master

**Comment by saw-jan** [comment]
> when i started this PR all changes were not present in stable so I did my PR in master

You can change the target now as well if you want. It would be easier if we maintain the flow

File: tests/e2e/support/objects/app-files/share/collaborator.ts (hunk 4 above)
Comment (amrita-shrestha) [reply tests/e2e/support/objects/app-files/share/collaborator.ts:121]: this work for me
```
const CUSTOM_PERMISSIONS: readonly CustomPermissionType[] = resourceType === 'folder' ? Collaborator.FOLDER_CUSTOM_PERMISSIONS : Collaborator.FILE_CUSTOM_PERMISSIONS
```

File: tests/e2e/support/objects/app-files/share/collaborator.ts [hunk 5]
```diff
@@ -126,6 +142,7 @@ export default class Collaborator {
   static async setCollaboratorRole(
     page: Page,
     role: string,
+    resourceType?: string,
```
[the remark this answers is not in the document]
Comment (amrita-shrestha) [reply tests/e2e/support/objects/app-files/share/collaborator.ts]: i don't think so

**Review by saw-jan — COMMENTED** [review]
Lookign good :+1:

**Review by SagarGi — COMMENTED** [review]
code LGTM :+1:  Please rebase it.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/31699/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31699/tracing/folder-alice-2023-1-17-07-51-54.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31699/tracing/participant-management-alice-2023-1-17-07-53-58.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31699/tracing/participant-management-anonymous-2023-1-17-07-54-30.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31699/tracing/participant-management-brian-2023-1-17-07-54-08.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31699/tracing/participant-management-carol-2023-1-17-07-54-19.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31699/tracing/participant-management-marie-2023-1-17-07-54-11.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31699/tracing/participant-management-richard-2023-1-17-07-54-15.zip 


</p></details>

**Review by ScharfViktor — APPROVED** [review]
LGTM

**Comment by saw-jan** [comment]
- [ ] port to stable-6 (https://github.com/owncloud/web/pull/8291)

**Pull request merged**